### DISCUSSION TITLE
How can a user define the `extraCloneOpts` for a given repository?

### DISCUSSION THREAD

**Discussion by jamietanna** [renovatebot/renovate#24985] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.3.3

### If you're self-hosting Renovate, select which platform you are using.

GitHub.com

### Was this something which used to work for you, and then stopped?

Never seen this work

### Wanted end result.

I notice that inside the utility classes we use to `git clone` the repo, there's [an ability to specify `extraCloneOpts](https://github.com/renovatebot/renovate/blob/0da50b51604ec918afbd5981ba8c834e03366484/lib/util/git/index.ts#L427-L432) which are [defined in these types](https://github.com/renovatebot/renovate/blob/0da50b51604ec918afbd5981ba8c834e03366484/lib/util/git/types.ts#L14-L34).

But I can't see any documentation for how to tune this, if it's even possible outside of [the platform itself](https://github.com/renovatebot/renovate/blob/0da50b51604ec918afbd5981ba8c834e03366484/lib/modules/platform/azure/index.ts#L233-L238) performing the clone?

### What you tried so far.

Just looking through source code.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by jamietanna** [comment]
I've implemented the extensibility in https://github.com/renovatebot/renovate/pull/24995 in the meantime - I'm not sure if this _is_ something we want to expose, but while playing around with it, I thought I'd at least get a PR raised

**Comment by rarkins** [comment]
I think today it's an internal option, utilized by Azure only

**Reply by jamietanna** [reply]
Thanks I spotted that while digging through - is this something we'd want to expose to consumers, or do you feel it's not necessary?

In my case, I'm looking to do i.e. `git clone --depth=1` when doing `--dry-run=lookup`, but that's not a super common case.

**Reply by rarkins** [reply]
depth=1 won't work though, for multiple reasons

**Reply by jamietanna** [reply]
Even for `extract`/`lookup` (presuming that `onboarding=false`)?

**Reply by rarkins** [reply]
For that it could be ok, but as soon as you want to write branches and check things like behind base branch, you need history

**Reply by viceice** [reply]
we already do a blobless clone via git filter, so we should only download the blond for the main branch until other branches are checked out.

so it should be a kind of light weight clone and depth=1 shouldn't improve much.

blobless clone depends on git backend, which should be supported by GitHub, gitea and gitlab. not sure about other platforms.
